pyramid/view.py

```diff
@@ -690,17 +699,23 @@ def invoke_exception_view(
             # we use .get instead of .__getitem__ below due to
             # https://github.com/Pylons/pyramid/issues/700
             request_iface = attrs.get('request_iface', IRequest)
-            response = _call_view(
-                registry,
-                request,
-                exc,
-                context_iface,
-                '',
-                view_types=None,
-                view_classifier=IExceptionViewClassifier,
-                secure=secure,
-                request_iface=request_iface.combined,
-                )
+
+            try:
+                manager.push({'request': request, 'registry': registry})
```

Comment:
This `push` should go before the `try`. You do not want to try to `pop` if the `push` fails. It's pedantic but could avoid raising an exception from an exception.